homeassistant/components/zwave_js/config_flow.py

```diff
@@ -630,6 +633,70 @@ async def async_step_configure_addon_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Ask for config for Z-Wave JS add-on."""
+
+        # For recommended installation, automatically set network type to "new"
+        if self._recommended_install and self._usb_discovery:
+            user_input = {"network_type": NETWORK_TYPE_NEW}
+
+        if user_input is not None:
+            if not self._usb_discovery:
+                self.usb_path = user_input[CONF_USB_PATH]
+
+            if user_input["network_type"] == NETWORK_TYPE_NEW:
+                # Set all keys to empty strings for new network
+                self.s0_legacy_key = ""
+                self.s2_access_control_key = ""
+                self.s2_authenticated_key = ""
+                self.s2_unauthenticated_key = ""
+                self.lr_s2_access_control_key = ""
+                self.lr_s2_authenticated_key = ""
+
+                addon_config_updates = {
+                    CONF_ADDON_DEVICE: self.usb_path,
+                    CONF_ADDON_S0_LEGACY_KEY: self.s0_legacy_key,
+                    CONF_ADDON_S2_ACCESS_CONTROL_KEY: self.s2_access_control_key,
+                    CONF_ADDON_S2_AUTHENTICATED_KEY: self.s2_authenticated_key,
+                    CONF_ADDON_S2_UNAUTHENTICATED_KEY: self.s2_unauthenticated_key,
+                    CONF_ADDON_LR_S2_ACCESS_CONTROL_KEY: self.lr_s2_access_control_key,
+                    CONF_ADDON_LR_S2_AUTHENTICATED_KEY: self.lr_s2_authenticated_key,
+                }
+
+                await self._async_set_addon_config(addon_config_updates)
+                return await self.async_step_start_addon()
+
+            # Network already exists, go to security keys step
+            return await self.async_step_configure_security_keys()
+
+        usb_path = self.usb_path or ""
+
+        schema: VolDictType = {
+            vol.Required("network_type", default=""): vol.In(
+                [NETWORK_TYPE_NEW, NETWORK_TYPE_EXISTING]
+            )
+        }
+
+        if not self._usb_discovery:
+            try:
+                ports = await async_get_usb_ports(self.hass)
+            except OSError as err:
+                _LOGGER.error("Failed to get USB ports: %s", err)
+                return self.async_abort(reason="usb_ports_failed")
+
+            schema = {
+                vol.Required(CONF_USB_PATH, default=usb_path): vol.In(ports),
+                **schema,
```

Comment:
Doesn't it look weird to ask for both the USB device and if the network is new in the same form? I'd consider separating those questions to different forms/steps.

Reply:
I was wondering the same. Then it can also be a menu. If I have time for it today I'll change it, if not I'll merge this and update later.